mcs/class/corlib/System.IO/DirectoryInfo.cs

```diff
@@ -424,39 +424,51 @@ public IEnumerable<FileSystemInfo> EnumerateFileSystemInfos (string searchPatter
 			return EnumerateFileSystemInfos (FullPath, searchPattern, searchOption);
 		}
 
-		static internal IEnumerable<FileSystemInfo> EnumerateFileSystemInfos (string full, string searchPattern, SearchOption searchOption)
+		static internal IEnumerable<FileSystemInfo> EnumerateFileSystemInfos (string basePath, string searchPattern, SearchOption searchOption)
 		{
-			string path_with_pattern = Path.Combine (full, searchPattern);
-			IntPtr handle = IntPtr.Zero;
-			MonoIOError error;
-			FileAttributes rattr;
-			bool subdirs = searchOption == SearchOption.AllDirectories;
+			Path.Validate (basePath);
+
+			IntPtr findHandle = IntPtr.Zero;
 
-			Path.Validate (full);
-			
 			try {
-				string s = MonoIO.FindFirst (full, path_with_pattern, out rattr, out error, out handle);
-				if (s == null)
-					yield break;
-				if (error != 0)
-					throw MonoIO.GetException (Path.GetDirectoryName (path_with_pattern), (MonoIOError) error);
+				string filePath;
+				int attrs;
+
+				try {} finally {
```

Comment:
What do you need this for?

Reply:
We don't want a `ThreadAbortException` to leak the `findHandle`, so to guarantee that a `Thread.Abort` doesn't happen there, we put it in a `finally` clause which will delay the `ThreadAbortException` until it exits.

Reply:
The BCL way of handling this is to use a SafeHandle.